scripts/utils/generate_vif/constants.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+XML_ENCODING = "utf-8"
+XML_ELEMENT_NAME_PREFIX = "vif"
+XML_ROOT_ELEMENT_NAME = "VIF"
+XML_NAMESPACE_ATTRIBUTES = {
+    "xmlns:opt": "http://usb.org/VendorInfoFileOptionalContent.xsd",
+    "xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
+    "xmlns:vif": "http://usb.org/VendorInfoFile.xsd",
+}
+
+NAME = "name"
+VALUE = "value"
+TEXT = "text"
+ATTRIBUTES = "attributes"
+CHILD = "child"
+COMPONENT = "Component"
+
+SINK_PDOS = "sink-pdos"
+SINK_PDO = "SnkPDO"
+SINK_PDO_SUPPLY_TYPE = "Snk_PDO_Supply_Type"
+SINK_PDO_VOLTAGE = "Snk_PDO_Voltage"
+SINK_PDO_OP_CURRENT = "Snk_PDO_Op_Current"
+SINK_PDO_MIN_VOLTAGE = "Snk_PDO_Min_Voltage"
+SINK_PDO_MAX_VOLTAGE = "Snk_PDO_Max_Voltage"
+SINK_PDO_OP_POWER = "Snk_PDO_Op_Power"
+PD_POWER_AS_SINK = "PD_Power_As_Sink"
+NUM_SINK_PDOS = "Num_Snk_PDOs"
+MODEL_PART_NUMBER = "Model_Part_Number"
+EPR_SUPPORTED_AS_SINK = "EPR_Supported_As_Snk"
+NO_USB_SUSPEND_MAY_BE_SET = "No_USB_Suspend_May_Be_Set"
+HIGHER_CAPABILITY_SET = "Higher_Capability_Set"
+FR_SWAP_REQD_TYPE_C_CURRENT_AS_INITIAL_SOURCE = "FR_Swap_Reqd_Type_C_Current_As_Initial_Source"
+
+VIF_SPEC_ELEMENTS = {
+    "VIF_Specification": {
+        TEXT: "3.19",
+    },
+    "VIF_App": {
+        CHILD: {
+            "Vendor": {
+                TEXT: "USB-IF",
+            },
+            "Name": {
+                TEXT: "VIF Editor",
+            },
+            "Version": {
+                TEXT: "3.2.4.0",
+            }
+        }
+    },
+    "Vendor_Name": {
+        TEXT: "Google",
```

Comment:
Google ?

Is this internal work that you just want upstreamed, or is this actually useful for the Zephyr community, and in which way ?

Reply:
I removed this element for this PR. I will be generating "Vendor_Name" element in xml from device tree, in next iteration